cli/src/main.rs

```diff
@@ -153,7 +163,11 @@ fn list_connected_devices() {
 }
 
 fn dump_memory(n: usize, loc: u32, words: u32) -> Result<(), CliError> {
-    with_device(n as usize, Box::new(probe::target::m0::M0::default()), |mut session| {
+    let target = probe::target::Target::new(
+        probe::target::m0::M0::default(),
+        probe::target::nrf51822::nRF51822(),
```

Comment:
This seems a bit superfluous, when we have a NRF51 we already now that it is a Cortex M0, so maybe we could just combine it?